src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantInterpreter.java

```diff
@@ -113,18 +121,153 @@ private static AssemblyContigWithFineTunedAlignments furtherPreprocess(final Ass
             //       if you are concerned about the first child alignment from the same gapped alignment being skipped,
             //       don't worry, that is impossible because child alignments of the same gapped alignment cannot overlap on the read.
             if (two.alnModType.equals(ContigAlignmentsModifier.AlnModType.FROM_SPLIT_GAPPED_ALIGNMENT)) {
-                final int overlapOnRead = AlignmentInterval.overlapOnContig(one, two);
-                if (overlapOnRead >= two.getSizeOnRead())
+                if ( one.containsOnRead(two) )
                     continue;
             }
-            final List<AlignmentInterval> deoverlapped = ContigAlignmentsModifier.removeOverlap(one, two, refSequenceDictionary);
-            result.add(deoverlapped.get(0));
-            one = deoverlapped.get(1);
+            final int overlapOnContig = AlignmentInterval.overlapOnContig(one, two);
+            if ( overlapOnContig == 0 ) { // nothing to remove
+                result.add(one);
+                one = two;
+            } else {
+                Boolean hintRequestYieldingOverlapToAlignmentTwo = null;
```

Comment:
Rather than a three value variable (`null`/`true`/`false`) which is effectively how you're using this, I think it would be more explicit if you just had two booleans, eg, `firstIsAlignmentHead`, `secondIsAlignmentTail`

Reply:
changed as suggested.